reactos/dll/win32/browseui/basebarsite.cpp

```diff
@@ -312,19 +388,42 @@ HRESULT STDMETHODCALLTYPE CBaseBarSite::IsWindowOwner(HWND hWnd)
 
 HRESULT STDMETHODCALLTYPE CBaseBarSite::OnFocusChangeIS (IUnknown *punkObj, BOOL fSetFocus)
 {
-    return E_NOTIMPL;
+    CComPtr<IInputObjectSite>               objectSite;
+    HRESULT                                 hResult;
+
+    hResult = fDeskBarSite->QueryInterface(IID_PPV_ARG(IInputObjectSite, &objectSite));
+    if(FAILED_UNEXPECTEDLY(hResult))
+        return hResult;
+
+    // FIXME: should we directly pass-through, or advertise ourselves as focus owner ?
```

Comment:
I don't really how to do an apitest here, since the class seems to be internal to browseui and not instanciable from the outside world (but I may have missed a magic undocumented CLSID :D).